### PR TITLE
feat(engine, api): metrics through REST API

### PR DESCRIPTION
**Pull request by fsamin**
usage:
```
curl  http://localhost:8081/mon/metrics/detail/
```
returns the list of all metrics as a JSON array

```
curl -vv http://localhost:8081/mon/metrics/detail/cds/nb_users
```
Returns the detail of a metric.
Here the metric is named `cds/nb_users`

```
curl "http://localhost:8081/mon/metrics/detail/cds/queue?range=all&status=building"
```
Returns the detail of a tagged metric

You can add the query parameter `threshold=x`
to raised a 509 HTTP Error if the threshold is reached of exceeded

Signed-off-by: francois  samin <francois.samin@corp.ovh.com>

### CODE REVIEW

File: engine/service/metrics.go [hunk 1]
```diff
@@ -198,3 +200,52 @@ func RegisterCommonMetricsView(ctx context.Context) {
...
+			}
+
+			statusCode := http.StatusOK
+			if threshold != "" {
+				thresholdF, err := strconv.ParseFloat(threshold, 64)
+				if err != nil {
+					return sdk.WithStack(sdk.ErrWrongRequest)
+				}
+				if metricsView.Value >= thresholdF {
+					log.Error(context.Background(), "GetMetricHandler> %s threshold (%s) reached or exceeded : %v", metricsView.Name, threshold, metricsView.Value)
```
Comment (richardlt) [inline engine/service/metrics.go]: log.Error(ctx

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12857.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✘  
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12857.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✘  
  * Test SDK Rust ✔ 
  * Test UI ✔

File: engine/service/metrics.go (hunk 1 above)
Comment (fsamin) [reply engine/service/metrics.go]: Well the error is not related to the http request

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12857.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12857.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

File: engine/service/metrics.go (hunk 1 above)
Comment (richardlt) [reply engine/service/metrics.go]: Ok

**Pull request merged**